### PR TITLE
Merge back 1.4.0 to develop

### PR DESCRIPTION
**Pull request by yunchu**
### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by eunwoosh — COMMENTED** [review]
When I checked my changes, they're applied well. Thanks.
By the way, there is a code conflict. Could you check it?

File: src/otx/__init__.py [hunk 1]
```diff
@@ -3,5 +3,5 @@
 # Copyright (C) 2021-2023 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
-__version__ = "1.4.0rc0"
+__version__ = "1.4.0rc4"
```
Comment (goodsong81) [inline src/otx/__init__.py:6]: When do we promote this to 1.5.0rc0? It should have been already at the time of release branch creation.

**Review by sungmanc — APPROVED** [review]
LGTM

File: src/otx/api/usecases/exportable_code/demo/requirements.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
 openvino==2023.0
-openvino-model-api==0.1.2
-otx @ git+https://github.com/openvinotoolkit/training_extensions/@77b635f4fba0a8acca221ec7e8b1fadd734358da#egg=otx
+openvino-model-api==0.1.3
+otx==1.4.0rc4
```
Comment (goodsong81) [inline src/otx/api/usecases/exportable_code/demo/requirements.txt:3]: This should be commit-based on develop.

**Review by goodsong81 — CHANGES_REQUESTED** [review]

File: src/otx/__init__.py (hunk 1 above)
Comment (yunchu) [reply src/otx/__init__.py:6]: I think if the merge back from the release branch could be done without having manual updates to the code, it's the best. so, i don't get it why the version should be promoted with release branch creation. can you put bit more you thoughts here?

Comment (goodsong81) [reply src/otx/__init__.py:6]: Current develop branch is actually otx==1.5.0rc0 but still is 1.4.0. When you create releases/1.4.0 the develop branch should be promoted to 1.5.0.

File: src/otx/api/usecases/exportable_code/demo/requirements.txt (hunk 2 above)
Comment (yunchu) [reply src/otx/api/usecases/exportable_code/demo/requirements.txt:3]: it's not easy to understand what I need to do for this PR. should it be the specific commit based on the develop branch? does it mean that should not be the version string?

Comment (goodsong81) [reply src/otx/api/usecases/exportable_code/demo/requirements.txt:3]: It was agreed so far that we maintain commit-based otx install on this line on develop branch.
Usually we set the latest commit of the PR. If you want someone who update `otx.api` to update this line later, please go ahead. It was just agreement, not strict rule.

**Review by goodsong81 — APPROVED** [review]

**Pull request merged**